### DISCUSSION TITLE
Failed to run inference in MONAI Label Server

### DISCUSSION THREAD

**Discussion by aleoneshahvar** [Project-MONAI/MONAILabel#515] [Q&A] [answered]
I am able to grab the next sample from my studies folder however this error pops up as soon as the image is grabbed:
Traceback (most recent call last):
  File "C:/Users/Ariela/AppData/Local/NA-MIC/Slicer 4.13.0-2021-10-03/NA-MIC/Extensions-30282/MONAILabel/lib/Slicer-4.13/qt-scripted-modules/MONAILabel.py", line 1549, in onClickSegmentation

    result_file, params = self.logic.infer(model, image_file, self.getParamsFromConfig("infer", model))

  File "C:/Users/Ariela/AppData/Local/NA-MIC/Slicer 4.13.0-2021-10-03/NA-MIC/Extensions-30282/MONAILabel/lib/Slicer-4.13/qt-scripted-modules/MONAILabel.py", line 1832, in infer

    result_file, params = client.infer(model, image_in, params, label_in)

  File "C:\Users\Ariela\AppData\Local\NA-MIC\Slicer 4.13.0-2021-10-03\NA-MIC\Extensions-30282\MONAILabel\lib\Slicer-4.13\qt-scripted-modules\MONAILabelLib\client.py", line 111, in infer

    "Status: {}; Response: {}".format(status, form),

MONAILabelLib.client.MONAILabelException: (2, "Status: 500; Response: b'Internal Server Error'")

Once I dismiss the error, I am still able to segment the image and submit it to the model. I can see that the model does in fact update and that the training bar progresses. I am unsure what this error means though and am wondering if it will cause errors down the line. Please let me know what you think!

**Comment by SachidanandAlle** [comment]
Please check the logs... either logs directory in your apps folder.. app.log in your app/logs folder
Or you can access online.. http://server:port/logs?lines=-1

After fetching the sample, slicer tries to run auto-segmentation (depending on the model/availability) and looks like it didn't get any label from server.... so you can check what happened at the server in its' logs

**Reply by aleoneshahvar** [reply]
Hello! Could you clarify what you mean by this? I've trained the model but when I go to test the automated segmentation, the same error appears. The link you provided also does not lead to anything. Thanks!

**Reply by aleoneshahvar** [reply]
What exactly am I looking for in the logs?

**Comment by SachidanandAlle** [comment]
Can you attach the log file?

**Reply by aleoneshahvar** [reply]
<img width="1372" alt="Screen Shot 2022-01-07 at 2 10 15 AM" src="https://user-images.githubusercontent.com/77142054/148506092-39273722-88cf-4060-b92e-3c45e9e0ba7d.png">
<img width="1374" alt="Screen Shot 2022-01-07 at 2 11 09 AM" src="https://user-images.githubusercontent.com/77142054/148506123-ca9b76b3-5c66-400e-b55d-c5bac9e38547.png">

<img width="1368" alt="Screen Shot 2022-01-07 at 2 11 23 AM" src="https://user-images.githubusercontent.com/77142054/148506135-6d590fac-fbe4-4d3a-9e0d-ac39cfecda03.png">

<img width="1372" alt="Screen Shot 2022-01-07 at 2 11 39 AM" src="https://user-images.githubusercontent.com/77142054/148506145-a72949ec-3cf2-4f5e-beb5-e02c6f1f52b0.png">

<img width="1373" alt="Screen Shot 2022-01-07 at 2 11 49 AM" src="https://user-images.githubusercontent.com/77142054/148506160-1d7e1119-1052-4f96-a975-af10ba403142.png">

Here are screenshots from the last entry in my logs!

**Comment by SachidanandAlle** [comment]
Do you have GPU with CUDA enabled?

**Reply by aleoneshahvar** [reply]
I don't believe I do, I get this runtime error:
<img width="1198" alt="Screen Shot 2022-01-07 at 1 22 35 AM" src="https://user-images.githubusercontent.com/77142054/148506776-73138253-8714-47f2-8579-9b7461bc2c62.png">

**Reply by SachidanandAlle** [reply]
It says CUDA not available

**Reply by aleoneshahvar** [reply]
Is there a way to correct this?

**Reply by SachidanandAlle** [reply]
Check if cuda enabled

python -c "import torch; print(torch.cuda.is_available())"

**Reply by SachidanandAlle** [reply]
If you NVIDIA gpu.. install CUDA.. and run the check again

**Reply by aleoneshahvar** [reply]
So I've successfully installed CUDA and it resolved the runtime error, however I'm still getting the "Failed to run inference on monai label server error". I've noticed now that this error pops up in my terminal and I'm not sure how to resolve it: ValueError: len(spatial_size) must be greater or equal to img spatial dimensions, got spatial_size=2 img=3.

**Reply by diazandr3s** [reply]
Hi @aleoneshahvar,

Thanks for sharing the update. Can you please tell us which App are you using? and which images are you using to run inference?
Please also share the latest logs you get after installing CUDA. It is easier for us to analyze from the logs.

**Reply by aleoneshahvar** [reply]
I'm using 3D Slicer and I'm using MR images of pregnant guinea pigs, the dataset is from my lab. I converted the DICOMs to .nii.gz format which has been working so far for the training process. Here are my logs post CUDA installation:
[Logs Post CUDA.txt](https://github.com/Project-MONAI/MONAILabel/files/7856858/Logs.Post.CUDA.txt)

**Reply by diazandr3s** [reply]
Thanks for this @aleoneshahvar. How many modalities have the images you're working with? It seems the error you've got comes from the conversion DICOM to nifti. My guess is that the images you're using are multimodality. Can you please check that?

**Reply by aleoneshahvar** [reply]
Do you know what the best way to view the modality is?

**Reply by aleoneshahvar** [reply]
When I load the images into ImageJ it says "wrong size, 512z104 expected, 512x512 found", I'm not sure if that helps at all!

**Reply by diazandr3s** [reply]
You can use Slicer or ITK-SNAP to check the image details (image size, modalities, spacing, etc) and also to visualize the images along with the labels. Currently, MONAI Label doesn't support multimodality datasets :/
Is it the dataset public? Would it be possible to share an image with their corresponding label? I'm happy to check them :)

**Reply by aleoneshahvar** [reply]
I'm having a hard time getting the files to attach here on github, do you have an email I can send them to by chance?

**Reply by diazandr3s** [reply]
andres.diaz-pinto@kcl.ac.uk

**Reply by lassoan** [reply]
> I'm having a hard time getting the files to attach here on github, do you have an email I can send them to by chance?

In general: upload anywhere (dropbox, onedrive, google drive, etc.) and post the link.

**Comment by SachidanandAlle** [comment]
@diazandr3s can you help..

**Comment by aleoneshahvar** [comment, marked as answer]
Just wanted to close this discussion! @diazandr3s was able to help me resolve the issues! The primary issue had to do with my manual segmentations. I used the paint tool on one of the older versions of MONAI Label to segment my dataset (pregnant guinea pigs), however, I segmented each 2D slice independently (this is how manual segmentations are done in my lab) and then submitted it to the model. Andres showed me how to conduct the segmentations using the 'grow from seeds' tool in 3D slicer and suggested that I update to the latest version of MONAI and slicer. Additionally, Andres showed me that the "Failed to run inference on MONAI Label server" error was popping up because I had "Auto-run pre-trained model" checked off under slicer's application settings (Edit -> Application Settings -> MONAI Label) while I am not using a pre-trained model. Thank you to everyone who responded and helped me with these issues!
